internal/provider/provider.go

```diff
@@ -40,14 +40,13 @@ const (
 func populateJujuProviderModelLive() (jujuProviderModel, error) {
 	data := jujuProviderModel{}
 	controllerConfig, err := juju.GetLocalControllerConfig()
-	if err != nil {
-		return data, err
-	}
-
 	data.ControllerAddrs = types.StringValue(getField(JujuControllerEnvKey, controllerConfig))
 	data.UserName = types.StringValue(getField(JujuUsernameEnvKey, controllerConfig))
 	data.Password = types.StringValue(getField(JujuPasswordEnvKey, controllerConfig))
 	data.CACert = types.StringValue(getField(JujuCACertEnvKey, controllerConfig))
+	if err != nil && !data.valid() {
```

Comment:
@Osama-Kassem for the third scenario above (where the CLI's not there, and half the env vars are empty), you're accurate that the behavior will be correct (i.e. we error), however, user will get the error `"the Juju CLI could not be accessed"`, which is inaccurate and possibly confusing because the user is trying to set things with the env vars and is not dealing with the juju cli anyways.

So I think changing the error message a bit would solve this. First I suggest changing the name of the `err` -> `cliNotExist` in the code, and then:

```suggestion
	if cliNotExist != nil && !data.valid() {
		return data, errors.New("Unable to acquire Juju controller config: no working Juju client, and environment variables are not fully set.")
	}
```

You can even change the `data.valid()` to return a slice of the faulty fields (or the names of those fields) to provide a better error message, but I think just a better err message would be sufficient for this PR.

Reply:
Done!